src/modules/twap/hooks/useCreateTwapOrder.ts

```diff
@@ -0,0 +1,46 @@
+import { useAtomValue } from 'jotai'
+import { useCallback } from 'react'
+
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+
+import ms from 'ms.macro'
+import { Nullish } from 'types'
+
+import { useAdvancedOrdersDerivedState } from 'modules/advancedOrders'
+import { useTradeConfirmActions } from 'modules/trade'
+
+import { useTwapOrderCreationContext } from './useTwapOrderCreationContext'
+
+import { settleTwapOrder } from '../services/settleTwapOrder'
+import { twapOrderAtom } from '../state/twapOrderAtom'
+
+export function useCreateTwapOrder() {
+  const twapOrder = useAtomValue(twapOrderAtom)
+
+  const { inputCurrencyAmount, outputCurrencyAmount } = useAdvancedOrdersDerivedState()
+
+  const twapOrderCreationContext = useTwapOrderCreationContext(inputCurrencyAmount as Nullish<CurrencyAmount<Token>>)
```

Comment:
What type does `inputCurrencyAmount` come as from `useAdvancedOrdersDerivedState`?

Cab we maybe update `useAdvancedOrdersDerivedState` or `useTwapOrderCreationContext` to work with the same type?

Reply:
`<CurrencyAmount<Currency>> !== <CurrencyAmount<Token>>` unfortunately.
The only difference is `Currency` can be `NativeCurrency | Token`, but in advanced order we don't use `NativeCurrency`

Reply:
Ah got it!